precompiles/distribution/tx.go

```diff
@@ -48,7 +48,7 @@ func (p Precompile) ClaimRewards(
 
 	// If the contract is the delegator, we don't need an origin check
 	// Otherwise check if the origin matches the delegator address
-	isContractDelegator := contract.CallerAddress == delegatorAddr
+	isContractDelegator := (contract.CallerAddress == delegatorAddr) && (origin != delegatorAddr)
```

Comment:
Why do we need an && here?

Reply:
is an extra condition to make sure the flag is correct.
The tx origin is always an EOA, so we need to check that `origin != delegatorAddr` to assert the contract is the delegator

Reply:
Oh yeah, it makes sense!